2024-01-24 20:20:24,437: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-01-24 20:20:31,591: root: INFO: Current backtesting datetime 2024-01-16 08:30:00-05:00
2024-01-24 20:20:31,591: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-24 20:20:31,591: root: INFO: Current backtesting datetime 2024-01-16 09:30:00-05:00
2024-01-24 20:20:31,591: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-24 20:20:31,591: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-24 20:20:31[0m
2024-01-24 20:20:31,591: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-24 20:20:31,620: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-24 20:20:31,620: root: ERROR: Strategy.get_bars() got an unexpected keyword argument 'start'
2024-01-24 20:20:31,620: root: ERROR: Traceback (most recent call last):
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 829, in run
    self._run_trading_session()
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 780, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 274, in func_output
    result = func_input(self, *args, **kwargs)
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 298, in func_output
    result = func_input(self, *args, **kwargs)
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 87, in func_output
    return func_input(self, *args, **kwargs)
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\strategies\strategy_executor.py", line 378, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
  File "C:\Users\Dell\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
  File "C:\Users\Dell\Documents\GitHub\bastard\bastard.py", line 141, in on_trading_iteration
    moving_average = self.calculate_moving_average(50)
  File "C:\Users\Dell\Documents\GitHub\bastard\bastard.py", line 124, in calculate_moving_average
    bars = self.get_bars('SPU', '1DAY', start=start_date, end=end_date, limit=days)
TypeError: Strategy.get_bars() got an unexpected keyword argument 'start'

2024-01-24 20:20:31,628: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-01-24 20:20:31,678: root: INFO: Backtesting finished
2024-01-24 20:20:31,678: root: WARNING: Cannot plot returns because the strategy returns are missing
2024-01-24 20:20:32,211: root: WARNING: Cannot create a tearsheet because the strategy returns are missing
2024-01-24 20:20:32,211: backtest_stats: INFO: Backtest took 0:00:07.789658 for a speed of 0.000
